public/app/plugins/datasource/influxdb/influxQLMetadataQuery.ts

```diff
@@ -0,0 +1,85 @@
+import { InfluxQueryTag } from './types';
+import InfluxDatasource from './datasource';
+import { InfluxQueryBuilder } from './query_builder';
+
+class InvalidMetadataError extends Error {
+  constructor(obj: unknown) {
+    console.error('invalid data', obj);
+    super('invalid data');
+  }
+}
+
+// to make sure the data returned from the server is the correct form.
+// TODO: remove this when we are finished (probably)
```

Comment:
Are you planning to remove it in this PR at the very end? If yes it might be worth adding a todo task (`[ ]`) in the description ;)

Reply:
good catch, yes, i will remove it :)